internal/bminventory/inventory.go

```diff
@@ -5641,3 +5642,35 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
 		return common.GenerateErrorResponder(common.NewApiError(http.StatusBadRequest, err))
 	}
 }
+
+func (b *bareMetalInventory) V2DownloadClusterFiles(ctx context.Context, params installer.V2DownloadClusterFilesParams) middleware.Responder {
+	if funk.Contains(clusterPkg.ClusterOwnerFileNames, params.FileName) {
+		authPayload := ocm.PayloadFromContext(ctx)
+		if ocm.UserRole != authPayload.Role {
+			errMsg := fmt.Sprintf("File '%v' is accessible only for cluster owners", params.FileName)
+			payload := common.GenerateInfraError(http.StatusForbidden, errors.New(errMsg))
+			return installer.NewV2DownloadClusterFilesForbidden().WithPayload(payload)
+		}
+	}
```

Comment:
We removed kubeconfig and admin-password from this API, so this code is no longer necessary.

Reply:
fixed